### PR TITLE
Update coding guidelines

### PR DESCRIPTION
**Pull request by alxmjo**
Update coding guidelines to reflect new organisation structure, in which algorithms are implemented as header files.

- [x] Read the [contribution guidelines][contrib-guidelines]

[contrib-guidelines]: https://github.com/faheel/Algos/blob/master/CONTRIBUTING.md
[unit-tests]: https://github.com/faheel/Algos/blob/master/C%2B%2B/UNIT_TESTS.md
[contents]: https://github.com/faheel/Algos/tree/master/C%2B%2B#contents

### CODE REVIEW

File: C++/CODING_GUIDELINES.md [hunk 1]
```diff
@@ -47,15 +47,14 @@ int pc_reader;             // Lots of things can be abbreviated "pc".
 int cstmr_id;              // Deletes internal letters.
 ```
 
-### File names
+### File names and locations
 - File and directory names should be all lowercase, with word separators (such as dashes and spaces) replaced with an underscore `_`.
-- All source files should have the extenstion `.cpp`, unit test files `.test.cpp`, and header files `.hpp`.
+- All source files should be located in `/include/<algorithm or data_structure>/<type>` and have the extension `.hpp`.
+- All unit test files should be located in `/test/<algorithm or data_structure>/<type>` and have the extension `.cpp`.
```
Comment (faheel) [inline C++/CODING_GUIDELINES.md]: Drop the leading `/` in the directory path.

File: C++/CODING_GUIDELINES.md [hunk 2]
```diff
@@ -47,15 +47,14 @@ int pc_reader;             // Lots of things can be abbreviated "pc".
...
 
-#### Examples
-- Source file: `insertion_sort.cpp`, `n_queens.cpp`
-- Unit test file: `fast_exponentiation.test.cpp`, `fibonacci.test.cpp`
-- Header file: `disjoint_set.hpp`, `binary_search_tree.hpp`
-- Directory: `number_theory`, `backtracking`
+#### Example
+- Source file: `lowest_common_subsequence.hpp` in `/include/algorithm/string`
+- Unit test file: `lowest_common_subsequence.cpp` in `/test/algorithm/string`
```
Comment (faheel) [inline C++/CODING_GUIDELINES.md]: Drop the leading `/` in the directory path.

**Review by faheel — DISMISSED** [review]

**Comment by alxmjo** [comment]
I'm away from my computer but will fix this Tuesday and resubmit.

**Comment by alxmjo** [comment]
Done.

**Pull request merged**